crates/sargon/src/types/hd_signature_input.rs

```diff
@@ -4,52 +4,46 @@ use crate::prelude::*;
 /// has the same signer, which would be a bug.
 #[derive(Clone, PartialEq, Eq, Hash, derive_more::Debug)]
 #[debug(
-    "HDSignatureInput {{ intent_hash: {:#?}, owned_factor_instance: {:#?} }}",
-    intent_hash,
+    "HDSignatureInput {{ payload_id: {:#?}, owned_factor_instance: {:#?} }}",
+    payload_id,
     owned_factor_instance
 )]
-pub struct HDSignatureInput {
+pub struct HDSignatureInput<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait?